From 1975393f0ee791bb184a9acfa1108c7578648c84 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Mon, 8 Oct 2018 13:33:11 -0400
Subject: [PATCH] initramfs: allow an optional wrapper script around initramfs
 generation

It is sometimes useful to have a secondary script call the
gen_initramfs_list.sh script. This allows special interpretation
of the initramfs root filesystem or other manipulations to be
injected.

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 usr/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/Makefile b/usr/Makefile
index 748f6a60bb1e..7dfd8d858244 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -24,7 +24,7 @@ $(obj)/initramfs_data.o: $(obj)/$(datafile_y) FORCE
 # Generate the initramfs cpio archive
 
 hostprogs-y := gen_init_cpio
-initramfs   := $(CONFIG_SHELL) $(srctree)/$(src)/gen_initramfs_list.sh
+initramfs   := $(INITRAMFS_WRAPPER) $(CONFIG_SHELL) $(srctree)/$(src)/gen_initramfs_list.sh
 ramfs-input := $(if $(filter-out "",$(CONFIG_INITRAMFS_SOURCE)), \
 			$(shell echo $(CONFIG_INITRAMFS_SOURCE)),-d)
 ramfs-args  := \
-- 
2.5.0

